sycl/plugins/level_zero/pi_level_zero.hpp

```diff
@@ -779,6 +784,37 @@ struct _pi_queue : _pi_object {
   // asked to not transfer the ownership to SYCL RT.
   bool OwnZeCommandQueue;
 
+  // Indicates if we create an event for each command or use eventless mode for
+  // the queue. The eventless mode is used for SYCL in-order queue with
+  // discard_events property. Since we don't create L0 events for some commands
```

Comment:
> The eventless mode is used for SYCL in-order queue with discard_events property.

Why is this a calculated member rather just taken from queue properties? What does it mean (if possible) for a PI queue to become not event-less where it started as event-less? Also, why is this only for "inorder"? Can't users want to do all of their sync of OOO queue directly on the queue, and not generate events?

Reply:
Yes, we can do that, I fixed it and now L0 Plugin takes it from queue property.  `discard_events` property can be used only for in-order queue.

Reply:
> discard_events property can be used only for in-order queue.

And I am asking why? I think we should be able to do so for OOO queues too.

Reply:
We need to check if we can do `discard_events` for OOO queue, if there are no restrictions from SYCL RT or other levels then we can extend this in the future, but for now, the first step is to provide this functionality for `in-order` queue.